Tella/Scenes/GDrive/ViewModel/GDriveOutboxViewModel.swift

```diff
@@ -0,0 +1,210 @@
+            uploadQueue.removeFirst()
+            uploadNextFile(folderId: folderId)
+            return
+        }
+        self.isFileLoading = false
+        currentUploadCancellable = gDriveRepository.uploadFile(fileURL: fileUrl, fileId: fileToUpload.id ?? "", mimeType: fileToUpload.mimeType ?? "", folderId: folderId)
+            .receive(on: DispatchQueue.main)
+            .sink(
+                receiveCompletion: { [weak self] completion in
+                    guard let self = self else { return }
+                    handleCompletionForUploadFile(completion, folderId: folderId)
+                },
+                receiveValue: { [weak self] progressInfo in
+                    self?.updateProgressInfos(uploadProgressInfo: progressInfo)
+                }
+            )
+    }
+    
+    private func handleCompletionForUploadFile(_ completion: Subscribers.Completion<APIError>, folderId: String) {
+        switch completion {
+        case .finished:
+            self.uploadQueue.removeFirst()
+            self.uploadNextFile(folderId: folderId)
+        case .failure( let error):
+            switch error {
+            default:
+                Toast.displayToast(message: error.errorDescription ?? error.localizedDescription)
+                updateReportStatus(reportStatus: .submissionError)
+            }
+        }
+    }
+    
+    private func handleCompletionForCreateFolder(_ completion: Subscribers.Completion<APIError>) {
+        switch completion {
+        case .finished:
+            break
+        case .failure( let error):
+            switch error {
+            default:
+                Toast.displayToast(message: error.errorDescription ?? error.localizedDescription)
+                updateReportStatus(reportStatus: .submissionError)
+            }
+        }
+    }
+    
+    override func pauseSubmission() {
+        if isSubmissionInProgress {
+            updateReportStatus(reportStatus: .submissionPaused)
+            gDriveRepository.pauseAllUploads()
+            currentUploadCancellable?.cancel()
+            currentUploadCancellable = nil
+        }
+    }
+    
+    override func updateReportStatus(reportStatus: ReportStatus) {
+        self.reportViewModel.status = reportStatus
+        
+        guard let id = reportViewModel.id else { return }
+        
+        mainAppModel.tellaData?.updateDriveReportStatus(reportId: id, status: reportStatus)
+    }
+    
+    private func updateReportFolderId(folderId: String) {
+        guard let id = reportViewModel.id else { return }
+        
+        mainAppModel.tellaData?.updateDriveFolderId(reportId: id, folderId: folderId)
+    }
+    
+    override func updateFileProgress(progressInfo:UploadProgressInfo) {
+        guard let reportId = reportViewModel.id else { return }
+        
+        let reportFiles = reportViewModel.files.map { file in
+            return ReportFile(
```

Comment:
Maybe it's easier to initiate it with` file: ReportVaultFile` and reportInstanceId: Int?

Reply:
for this case particularly yes, it's cleaner this way. But we are using ReportFIle in multiple places where we don't have the complete `ReportVaultFile`. What do you think of adding two initializers?

```
 init(id: Int? = nil,
         fileId: String? = nil,
         status: FileStatus? = nil,
         bytesSent: Int? = 0,
         createdDate: Date? = nil,
         updatedDate: Date? = Date(),
         reportInstanceId: Int? = nil) {
        self.id = id
        self.fileId = fileId
        self.status = status
        self.bytesSent = bytesSent
        self.createdDate = createdDate
        self.updatedDate = updatedDate
        self.reportInstanceId = reportInstanceId
    }
    
    init(file: ReportVaultFile, reportInstanceId: Int?) {
        self.id = file.instanceId
        self.fileId = file.id
        self.status = file.status
        self.bytesSent = file.bytesSent
        self.createdDate = file.createdDate
        self.updatedDate = file.updatedDate
        self.reportInstanceId = reportInstanceId
    }
    ````

Reply:
yes it's better add two initializers